src/vm.rs

```diff
@@ -1037,36 +1042,57 @@ impl<'a, E: UserDefinedError, I: InstructionMeter> EbpfVm<'a, E, I> {
+                            resolved = true;
+
+                            if instruction_meter_enabled {
+                                let _ = instruction_meter.consume(self.last_insn_count);
+                            }
+                            total_insn_count += self.last_insn_count;
+                            self.last_insn_count = 0;
+                            let mut result: ProgramResult<E> = Ok(0);
+                            (unsafe { std::mem::transmute::<u64, SyscallFunction::<E, *mut u8>>(syscall.function) })(
+                                self.syscall_context_objects[SYSCALL_CONTEXT_OBJECTS_OFFSET + syscall.context_object_slot],
+                                reg[1],
+                                reg[2],
+                                reg[3],
+                                reg[4],
+                                reg[5],
+                                &self.memory_mapping,
+                                &mut result,
+                            );
+                            reg[0] = result?;
+                            if instruction_meter_enabled {
+                                remaining_insn_count = instruction_meter.get_remaining();
+                            }
                         }
-                        total_insn_count += self.last_insn_count;
-                        self.last_insn_count = 0;
-                        let mut result: ProgramResult<E> = Ok(0);
-                        (unsafe { std::mem::transmute::<u64, SyscallFunction::<E, *mut u8>>(syscall.function) })(
-                            self.syscall_context_objects[SYSCALL_CONTEXT_OBJECTS_OFFSET + syscall.context_object_slot],
-                            reg[1],
-                            reg[2],
-                            reg[3],
-                            reg[4],
-                            reg[5],
-                            &self.memory_mapping,
-                            &mut result,
-                        );
-                        reg[0] = result?;
-                        if instruction_meter_enabled {
-                            remaining_insn_count = instruction_meter.get_remaining();
+                    }
+
+                    if calls {
+                        if let Some(target_pc) = self.executable.lookup_bpf_function(insn.imm as u32) {
+                            resolved = true;
+
+                            // make BPF to BPF call
+                            reg[ebpf::FRAME_PTR_REG] =
+                                self.stack.push(&reg[ebpf::FIRST_SCRATCH_REG..ebpf::FIRST_SCRATCH_REG + ebpf::SCRATCH_REGS], next_pc)?;
+                            next_pc = self.check_pc(pc, target_pc)?;
+                        }
+                    }
+
+                    if !resolved {
+                        if self.executable.get_config().disable_unresolved_symbols_at_runtime {
```

Comment:
This is an error branch anyway but for consistency: `config.disable_unresolved_symbols_at_runtime`